ambry-api/src/main/java/com/github/ambry/account/Dataset.java

```diff
@@ -228,3 +232,30 @@ public String toString() {
         + "]";
   }
 }
+
+/*
+Customize the serializer to fileter out all the null(default) value.
+ */
+class DatasetSerializer extends JsonSerializer<Dataset> {
+  @Override
+  public void serialize(Dataset dataset, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+    gen.writeStartObject();
+
+    // Use reflection to get all fields of the Dataset class
+    Field[] fields = Dataset.class.getDeclaredFields();
+    for (Field field : fields) {
+      if (field.isAnnotationPresent(JsonProperty.class)) {
+        try {
+          field.setAccessible(true);
+          Object value = field.get(dataset);
+          if (value != null) {
+            gen.writeObjectField(field.getName(), value);
```

Comment:
we should probably use annotation value here. something like
```
gen.writeObjectField(field.getAnnotation(JsonProperty.class).value(), value);
```

Reply:
Sure, updated it.